internal/service/project/resource_project_test.go

```diff
@@ -95,10 +94,11 @@ func TestGetProjectPropsFromAPI(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			testObject := new(MockProjectService)
-			testObject.On("ListProjectTeams", mock.Anything, mock.Anything).Return(tc.teamRoleReponse)
-			testObject.On("ListProjectLimits", mock.Anything, mock.Anything).Return(tc.limitResponse)
-			testObject.On("GetProjectSettings", mock.Anything, mock.Anything).Return(tc.groupResponse)
+			testObject := mocksvc.NewGroupProjectService(t)
+			testObject.On("ListProjectTeams", mock.Anything, mock.Anything).Return(tc.teamRoleReponse.ProjectTeamResp, tc.teamRoleReponse.HTTPResponse, tc.teamRoleReponse.Err).Maybe()
+			testObject.On("ListProjectLimits", mock.Anything, mock.Anything).Return(tc.limitResponse.LimitsResponse, tc.limitResponse.HTTPResponse, tc.limitResponse.Err).Maybe()
```

Comment:
knitpit: tc.limitResponse.LimitsResponse etc. seems a bit verbose

Reply:
I agree, but the alternative also seems quite verbose: removing the `ProjectResponse` attributes and adding 3 attributes for each one we remove, so instead of 3 attributes we would have 9:
```
teamRoleReponse ProjectResponse
limitResponse   ProjectResponse
groupResponse   ProjectResponse
```
compared to 
```
teamRoleResponse *admin.PaginatedTeamRole
teamRoleHTTPResponse   *http.Response
teamRoleError   error
limitResponse []admin.DataFederationLimit
limitHTTPResponse   *http.Response
limitError   error
groupSettingsResponse *admin.GroupSettings
limitHTTPResponse   *http.Response
limitError   error
```